powerstore/resource_volume.go

```diff
@@ -343,8 +395,17 @@ func (r resourceVolume) Read(ctx context.Context, req tfsdk.ReadResourceRequest,
 		)
 		return
 	}
+	// Get Volume Group Mapping details from API
+	volGroupMapping, err := r.p.client.PStoreClient.GetVolumeGroupsByVolumeID(context.Background(), volID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error fetching volume host mapping",
+			"Could not create volume, unexpected error: "+err.Error(),
+		)
+		return
+	}
 
-	updateVolState(&state, volResponse, hostMapping, nil, "Read")
+	updateVolState(&state, volResponse, hostMapping, volGroupMapping, nil, "Read")
```

Comment:
same as above
try to use https://github.com/dell/terraform-provider-powerstore/blob/dev/powerstore/defaults.go#L14

also, try to make updateVolState as method instead of function